apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ShufflingParticipant.java

```diff
@@ -20,199 +20,55 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
-import java.util.Arrays;
 
 @Slf4j
-public final class ShufflingParticipant {
-
-    public static final LinkKeyFactory<ShufflingData> shufflingDataDbKeyFactory = new LinkKeyFactory<ShufflingData>("shuffling_id", "account_id") {
-
-        @Override
-        public DbKey newKey(ShufflingData shufflingData) {
-            return shufflingData.getDbKey();
-        }
-
-    };
-    private static final Listeners<ShufflingParticipant, Event> listeners = new Listeners<>();
-    private static final LinkKeyFactory<ShufflingParticipant> shufflingParticipantDbKeyFactory = new LinkKeyFactory<ShufflingParticipant>("shuffling_id", "account_id") {
-
-        @Override
-        public DbKey newKey(ShufflingParticipant participant) {
-            return participant.dbKey;
-        }
-
-    };
-    private static final VersionedDeletableEntityDbTable<ShufflingParticipant> shufflingParticipantTable = new VersionedDeletableEntityDbTable<ShufflingParticipant>("shuffling_participant", shufflingParticipantDbKeyFactory) {
-
-        @Override
-        public ShufflingParticipant load(Connection con, ResultSet rs, DbKey dbKey) throws SQLException {
-            return new ShufflingParticipant(rs, dbKey);
-        }
-
-        @Override
-        public void save(Connection con, ShufflingParticipant participant) throws SQLException {
-            participant.save(con);
-        }
-
-    };
-    private static final PrunableDbTable<ShufflingData> shufflingDataTable = new PrunableDbTable<>("shuffling_data", shufflingDataDbKeyFactory) {
-        @Override
-        public boolean isScanSafe() {
-            return false; // shuffling data cannot be recovered from transactions (only by downloading/generating blocks)
-        }
-
-        @Override
-        public ShufflingData load(Connection con, ResultSet rs, DbKey dbKey) throws SQLException {
-            return new ShufflingData(rs, dbKey);
-        }
-
-        @Override
-        public void save(Connection con, ShufflingData shufflingData) throws SQLException {
-            try (
-                @DatabaseSpecificDml(DmlMarker.SET_ARRAY)
-                PreparedStatement pstmt = con.prepareStatement(
-                    "INSERT INTO shuffling_data (shuffling_id, account_id, data, "
-                        + "transaction_timestamp, height) "
-                        + "VALUES (?, ?, ?, ?, ?)")
-            ) {
-                int i = 0;
-                pstmt.setLong(++i, shufflingData.getShufflingId());
-                pstmt.setLong(++i, shufflingData.getAccountId());
-                DbUtils.setArrayEmptyToNull(pstmt, ++i, shufflingData.getData());
-                pstmt.setInt(++i, shufflingData.getTransactionTimestamp());
-                pstmt.setInt(++i, shufflingData.getHeight());
-                pstmt.executeUpdate();
-            }
-        }
-
-    };
+public final class ShufflingParticipant extends DerivedEntity {
```

Comment:
Looks like that entity is not at right package.
It should be inside package:
`com.apollocurrency.aplwallet.apl.core.entity.state.shuffling`
??